macros/PGauxiliaryFunctions.pl

```diff
@@ -107,72 +107,38 @@ sub Round {
 
 #least common multiple
 #VS 6/29/2000
+# made recursive to handle the gcf of any number of inputs 5/16/2020
 # ^function lcm
 sub lcm {
-	my $a = shift;
-	my $b = shift;
-
-	#reorder such that $a is the smaller number
-	if ($a > $b) {
-		my $temp = $a;
-		$a = $b;
-		$b = $temp;
-	}
-
-	my $lcm = 0;
-	my $curr = $b;;
-
-	while($lcm == 0) {
-		$lcm = $curr if ($curr % $a == 0);
-		$curr += $b;
-	}
-
-	$lcm;
-
+        if (scalar @_ == 0) {warn 'Cannot take lcm of the empty set'; return;}
```

Comment:
There is no need to use "scalar" here.  When comparing a list to a scalar it is converted to a scalar that is the number of elements in the list by default.

Reply:
This I do know, but I like using `scalar` for explicit code readability.